### DISCUSSION TITLE
MAUI port

### DISCUSSION THREAD

**Discussion by pablocity** [shinyorg/shiny#649] [Q&A] [answered]
Hi, I'd like to ask if there are any plans to port this framework to MAUI? Especially the BLE part, because MS stated that there are no plans to include BLE functionality in MAUI so it must be delivered by third party libraries and yours is one of the few.
Thanks

**Comment by aritchie** [comment, marked as answer]
It doesn't need to be ported.  It only needs to be shimmed into MAUI.  You can follow it here https://github.com/shinyorg/shiny/issues/580

**Reply by mbhoek** [reply]
Apologies if I'm asking for something obvious, but what exactly do you mean by "shimming it into MAUI"?
We're prototyping a .NET MAUI application that communicates with BLE devices, so I'm interested in how this would work.
Is there anything I can do to help, try out or maybe even contribute?

**Reply by aritchie** [reply]
I don't know all of the work involved yet since MAUI is still several months away from release.  When things begin to stabilize I will figure out all of the integration points with MAUI.  MAUI is a UI framework, Shiny is not, thus Shiny will work fine with or without MAUI.  Shiny was meant to run outside all of the UI so it only needs a partial integration with MAUI.  

As mentioned, follow the issue above and be patient.  It isn't a high priority at the moment.
